src/esm_parser/provenance.py

```diff
@@ -523,6 +523,23 @@ def get_provenance(self, index=-1):
 
         return provenance_dict
 
+    def get_first_provenance(self):
+        """
+        Recursively loops through the dictionary keys and returns the first provenance
+        found in the nested values.
+
+        Returns
+        -------
+        first_provenance : esm_parser.provenance.Provenance
+            The first provenance found in the nested values
+        """
+        first_provenance = None
+        for key, val in self.items():
+            if isinstance(val, PROVENANCE_MAPPINGS):
+                return val.get_first_provenance()
+            elif hasattr(val, "provenance"):
+                return val.provenance[-1]
```

Comment:
I would rename this to "oldest" since you getting the last element of the list (the one that was set at the very beginning). First is a bit confusing here, since I would intuitively expect you to get back `val.provenance[0]`

Reply:
it gives you the last provenance of the first value it finds in a list or a dictionary, that's why I wrote first. But I'll find a better name

Reply:
a fair point is ChatGPTs answer:

> what does it matter? The provenance you seek is just another step in an endless cycle of function calls and data structures, a fleeting glimpse of order in an otherwise chaotic and indifferent universe.